File: 2236-MaximumTwinSumOfALinkedList/2236-MaximumTwinSumOfALinkedList.cpp
// Last updated: 7/24/2025, 8:24:56 AM
class Solution {
public:
    int pairSum(ListNode* head) {
        ios_base::sync_with_stdio(false);
        cin.tie(nullptr);
        cout.tie(nullptr);
        int max=INT_MIN;
        vector <int> dq;
        ListNode *ptr=head;
        int n{0};
        while(ptr)
        {
            dq.push_back(ptr->val);
            ptr=ptr->next;
            n++;
        }
        for(int i=0;i<n/2;i++)
        {
            int sum = dq[i]+dq[n-i-1];
            if(sum>max)
                max=sum;
        }
        return max;
    }
};

File: 2236-MaximumTwinSumOfALinkedList/2236-MaximumTwinSumOfALinkedList.py
# Last updated: 7/24/2025, 8:24:49 AM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:

        fast = head
        prev = None
        curr = head
        
        while fast:
            temp = curr 
            fast=fast.next.next
            curr = temp.next
            temp.next = prev
            prev = temp
        res = float('-inf')
        while curr and prev:
            total = curr.val + prev.val
            res = max(total,res)
            curr = curr.next
            prev = prev.next

        return res